src/lib.rs

```diff
@@ -329,6 +329,16 @@ impl Config {
     /// otherwise cargo will link against the specified library.
     ///
     /// The given library name must not contain the `lib` prefix.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// gcc::Config::new()
+    ///             .file("src/foo.c")
+    ///             .shared_flag(true)
+    ///             .cpp_set_stdlib(Some("stdc++"))
+    ///             .compile("libfoo.so");
```

Comment:
Oh well so this option is intended for the C++ standard a piece of code is compiled with, but `cpp_set_stdlib` and such aren't really relevant for linkage other than telling rustc it may want a different C++ standard library.

Perhaps the `shared_flag` could be removed and `libfoo.so` could be changed to `libfoo.a`?